File: android/src/main/kotlin/com/ovaom/flutter_libpd/pd_controller/pd_message_receiver/PdMessageReceiver.kt
package com.ovaom.flutter_libpd.pd_controller.pd_message_receiver

interface PdMessageReceiver {
    fun addListener(source: String, callback: ListenerCallback)
    fun removeListener(source: String)
    fun removeAllListeners()
    fun dispose()
}

typealias ListenerCallback = (ReceivedEvent) -> Unit

data class ReceivedEvent(val source: String, val symbol: String? = null, val arguments: List<Any>? = null)

File: ios/Classes/PdController/PdMessageReceiver/PdMessageReceiver.swift
//
//  PdMessageReceiver.swift
//  flutter_libpd
//
//  Created by flav on 01/06/2022.
//

import Foundation

public protocol PdMessageReceiver {
    func addListener(source: String, callback: @escaping ListenerCallback)
    func removeListener(source: String)
    func removeAllListeners()
    func dispose()
}

public typealias ListenerCallback = (ReceivedEvent) -> Void

public struct ReceivedEvent {
    public let source: String
    public let symbol: String?
    public let arguments: [Any]?
    
    public init(source: String, symbol: String? = nil, arguments: [Any]? = nil) {
        self.source = source
        self.symbol = symbol
        self.arguments = arguments
    }
}
